lib/charms/hpc_libs/v0/is_container.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Detect if machine is a container instance.
+
+Even though Juju supports using LXD containers as the backing cloud for
+deploying charmed operators, not all HPC applications work within system containers,
+and some need additional configuration. This simple charm library provides the `is_container`
+function, a simple, deterministic way to identify if the charm is deployed into
+a system container using `systemd-detect-virt`.
+
+Exit code 0 means that the charm is running with a container. A non-zero exit code
+means that the charm is not running within a container.
```

Comment:
Maybe we could skip mentioning the internals of the library? It would be better so that we don't need to change the documentation if the implementation changes.

Reply:
Sure sure, I can drop it. `returncode == 0` already implies that an exit code of 0 means that we are running inside of a container.